app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala

```diff
@@ -205,12 +295,60 @@ class RepresenteeController @Inject() (
+              val response: EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]] =
+                id match {
+                  case RepresenteeNino(ninoId) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Right(ninoId), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeSautr(sautr) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Left(sautr), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeCgtReference(cgtReference) => {
+                    subscriptionService.getSubscribedDetails(cgtReference).map(sd => Some(Right(sd)))
+                  }
+                  case NoReferenceId =>
+                    EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]](
+                      Future.successful(Right(None))
+                    )
+                  case _ =>
+                    EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]](
+                      Future.successful(Left(Error("error")))
+                    )
+                }
+
+              val nameCheck: EitherT[Future, Error, Unit] = response.subflatMap(value =>
+                value match {
+                  case Some(Left(BusinessPartnerRecordResponse(_, _))) =>
+                    Right()
+                  case Some(Right(SubscribedDetails(Right(i), _, _, _, _, _, _)))
+                      if name.fold(false)(doNamesMatch(_, i)) =>
+                    Right()
+                  case Some(Right(SubscribedDetails(Left(_), _, _, _, _, _, _))) =>
+                    Right()
+                  case None => Right()
+                  case _ => {
+                    Left(Error(s"Name check error"))
+                  }
+                }
+              )
+
               val newAnswers = answers.fold(_.copy(id = Some(id)), _.copy(id = id))
 
-              updateDraftReturnAndSession(newAnswers, journey).fold({ e =>
-                logger.warn("Could not update draft return", e)
-                errorHandler.errorResult()
-              }, _ => Redirect(routes.RepresenteeController.checkYourAnswers()))
+              val result = for {
+                _ <- nameCheck.leftMap { e =>
+                      logger.warn("Name check error", e)
+                      BadRequest("Name check error")
+                    }
+                _ <- updateDraftReturnAndSession(newAnswers, journey).leftMap { e =>
+                      logger.warn("Could not update draft return", e)
+                      errorHandler.errorResult()
+                    }
+              } yield ()
+
+              result.fold(e => e, _ => Redirect(routes.RepresenteeController.confirmPerson()))
```

Comment:
can we get a redirect to the cya endpoint here? What you will probably need to get this to work is a boolean flag in the incomplete answers somewhere to tell you whether or not the person has confirmed the person. If a user closes the browser and comes back to the return, the cya endpoint needs to know whether to show the confirm person page or the check contact details page

Reply:
Done.